extensions/amp-ad-network-adsense-impl/0.1/test/test-responsive-state.js

```diff
@@ -0,0 +1,248 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  ADSENSE_MCRSPV_TAG,
+  ADSENSE_RSPV_TAG,
+  ADSENSE_RSPV_WHITELISTED_HEIGHT,
+} from '../../../../ads/google/utils';
+import {MAX_HEIGHT_EXP, ResponsiveState} from '../responsive-state';
+import {Services} from '../../../../src/services';
+import {
+  addAttributesToElement,
+  createElementWithAttributes,
+} from '../../../../src/dom';
+import {forceExperimentBranch} from '../../../../src/experiments';
+import {layoutRectLtwh} from '../../../../src/layout-rect';
+
+describes.realWin(
+  'responsive-state',
+  {
+    amp: {
+      extensions: ['amp-ad', 'amp-ad-network-adsense-impl'],
```

Comment:
i guess you don't need those extensions in this unit test.

Reply:
done